jubilant/_juju.py

```diff
@@ -837,6 +831,135 @@ def scp(
+                name=obj.get('name', None),
+                owner=obj.get('owner', '<model>'),
+                rotation=obj.get('rotation', 'never'),
+                revision=obj.get('revision', 1),
+                description=obj.get('description', ''),
+                created=datetime.datetime.fromisoformat(
+                    str(obj.get('created', '').replace('Z', '+00:00'))
+                ),
+                updated=datetime.datetime.fromisoformat(
+                    str(obj.get('updated', '').replace('Z', '+00:00'))
+                ),
+                content=None,
+                checksum='',
+                expires=None,
+                rotates=None,
+                label=None,
+                error='',
+                access=[],
+                revisions=[],
+            )
+            for uri_from_juju, obj in output.items()
+        ]
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: Literal[True],
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Revealed]: ...
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: Literal[False] = False,
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Hidden]: ...
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: Literal[False] = False,
+        revision: None = None,
+        revisions: Literal[True],
+    ) -> Secret[Hidden]: ...
+
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: bool = False,
+        revision: int | None = None,
+        revisions: bool = False,
+    ) -> Secret[Revealed] | Secret[Hidden]:
+        """Get the content of a secret.
+
+        Args:
+            name_or_uri: Name or URI of the secret to reveal.
+            reveal: Whether to reveal the secret content.
+            revisions: Whether to include all revisions of the secret. Mutually
+                exclusive with `reveal` and `revision`.
+            revision: Optional revision number of the secret to reveal. If not specified,
+                the latest revision is revealed.
+        """
+        args = ['show-secret', name_or_uri, '--format', 'json']
+        if reveal:
+            args.append('--reveal')
+        if revisions:
+            args.append('--revisions')
+        if revision is not None:
+            args.extend(['--revision', str(revision)])
+        stdout = self.cli(*args)
+        output: dict[str, SecretResponse] = json.loads(stdout)
+        uri_from_juju, obj = next(iter(output.items()))
+        return Secret[Any](
+            uri=SecretURI('secret:' + uri_from_juju),
+            revision=obj.get('revision', 0),
+            checksum=obj.get('checksum', ''),
+            owner=obj.get('owner', '<model>'),
```

Comment:
Same as above, I'm sure this is always provided.
```suggestion
            owner=obj['owner'],
```